classes/external/enqueue_merge_request.php

```diff
@@ -0,0 +1,120 @@
+                'keepuservalue' => new external_value(PARAM_RAW, 'Keep user value'),
+            ]
+        );
+    }
+    /**
+     * Return identifier of the enqueue merge request
+     *
+     * @param $removeuserid
+     * @param $removeuservalue
+     * @param $keepuserid
+     * @param $keepuservalue
+     */
+    public static function execute(string $removeuserfield,
+                                     string $removeuservalue,
+                                   string $keepuserfield,
+                                   string $keepuservalue): int {
+        // Validate all of the parameters.
+        $params = self::validate_parameters(self::execute_parameters(),
+                                            ['removeuserfield' => $removeuserfield,
+                                            'removeuservalue' => $removeuservalue,
+                                            'keepuserfield' => $keepuserfield,
+                                            'keepuservalue' => $keepuservalue
+                                            ]);
+        global $DB;
+        // Insert of the merge request into mdl_merge_request_queue Moodle table.
+        $usertoremove = $DB->get_records(merge_request::TABLE_USERS,
+                                        [$removeuserfield => $removeuservalue]);
+        if (count($usertoremove) == 0) {
+            throw new Exception(get_string('cannotfindusertoremove', 'tool_mergeusers'));
+        }
+        if (count($usertoremove) > 1) {
+            throw new Exception(get_string('toomanyuserstoremovefound', 'tool_mergeusers'));
+        }
+        foreach ($usertoremove as $item) {
+            $removeuserid = $item->id;
+        }
+        // Verify user to keep.
+        $usertokeep = $DB->get_records(merge_request::TABLE_USERS,
+                                        [$keepuserfield => $keepuservalue]);
+        if (count($usertokeep) == 0) {
+            throw new Exception(get_string('cannotfindusertokeep', 'tool_mergeusers'));
+        }
+        if (count($usertokeep) > 1) {
+            throw new Exception(get_string('toomanyuserstokeepfound', 'tool_mergeusers'));
+        }
+        foreach ($usertokeep as $item) {
+            $keepuserid = $item->id;
+        }
+        $timeadded = time();
+        $status = merge_request::QUEUED_NOT_PROCESSED;
+        $retries = 0;
+        $idrecord = $DB->insert_record(
+            merge_request::TABLE_MERGE_REQUEST,
+            [
+                'removeuserfield' => $removeuserfield,
+                'removeuservalue' => $removeuservalue,
+                'removeuserid' => $removeuserid,
+                'keepuserfield' => $keepuserfield,
+                'keepuservalue' => $keepuservalue,
+                'keepuserid' => $keepuserid,
+                'timeadded' => $timeadded,
+                'status' => $status,
+                'retries' => $retries
+            ],
+            $returnid = true,
+            $bulk = false
```

Comment:
you can omit default values from `insert_record`, like $bulk.

Reply:
obmitted the two:
 $returnid = true,
            $bulk = false